File: 125-ValidPalindrome/125-ValidPalindrome.java
// Last updated: 7/13/2026, 10:12:52 AM
class Solution {
    public boolean isPalindrome(String s) {
       
        StringBuilder sb = new StringBuilder();
        for(char c : s.toCharArray()) {
            if(Character.isLetterOrDigit(c)) {
                sb.append(Character.toLowerCase(c));
            }
        }


        String filtered = sb.toString();
        String reversed = sb.reverse().toString();

        return filtered.equals(reversed);
    }
}

File: 125-ValidPalindrome/125-ValidPalindrome.py
# Last updated: 7/13/2026, 10:13:02 AM
class Solution:
    def isPalindrome(self, s: str) -> bool:
        a=""
        for i in s:
            if i in 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890':
                a+=i.lower()
        if a == a[::-1]:
            return True
        else:
            return False        